docs/source/guides/speed.rst

```diff
@@ -0,0 +1,279 @@
+.. testsetup:: *
+
+    from pytorch_lightning.trainer.trainer import Trainer
+    from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+    from pytorch_lightning.core.lightning import LightningModule
+
+.. _speed:
+
+#######################
+Speed up model training
+#######################
+
+There are multiple ways you can speed up your model.
```

Comment:
This needs a little more expostion

Reply:
Add a table of contents with short links to the relevant part of the tutorial

Reply:
I feel like we are missing a lot of flags and tips that can help speed up convergence. The ones listed in the notion were just examples let's make sure we are being exhaustive. These should be living documents that we can update as we add new features as well so that the docs become less like long restaurant menus of features but rather more accessible tutorials.